src/app/views/query-runner/request/permissions/TabList.tsx

```diff
@@ -52,25 +62,27 @@ const TabList = ({ columns, classes, renderItemColumn, renderDetailsHeader, maxH
     return displayNotSignedInMessage();
   }
 
-
+  if (filteredPermissions.length === 0) {
+    return displayNoPermissionsFoundMessage();
+  }
 
   return (
     <>
       <Label className={classes.permissionLength}>
-        <FormattedMessage id='Permissions' />&nbsp;({permissions.length})
+        <FormattedMessage id='Permissions' />&nbsp;({filteredPermissions.length})
       </Label>
       <Label className={classes.permissionText}>
-        {!tokenPresent && <FormattedMessage id='sign in to consent to permissions' />}
-        {tokenPresent && <FormattedMessage id='permissions required to run the query' />}
+        {!tokenPresent && <FormattedMessage id={permissionModeType ? 'sign in to consent to permissions' : 'sign in to consent to application permissions'} />}
+        {tokenPresent && <FormattedMessage id={permissionModeType ? 'permissions required to run the query' : 'application permissions required to run the query'} />}
```

Comment:
```suggestion
        {!tokenPresent 
            && <FormattedMessage id={permissionModeType 
                ? 'sign in to consent to permissions' 
                : 'sign in to consent to application permissions'} />}
        {tokenPresent 
            && <FormattedMessage id={permissionModeType 
                ? 'permissions required to run the query' 
                : 'application permissions required to run the query'} />}
```
I think these lines started to get a bit too long without breaks.